src/WattleScript.Interpreter/Tree/Statements/ForEachLoopStatement.cs

```diff
@@ -164,6 +172,66 @@ public override void Compile(FunctionBuilder bc)
 			bc.PopSourceRef();
 		}
 
+		private bool CompilePossibleLiteralRange(Loop l, FunctionBuilder bc)
+		{
+			if (!(m_RValues is ExprListExpression listExpr)) return false;
+			if (listExpr.expressions.Count != 1) return false;
+			
+			Expression expr = listExpr.expressions[0];
+			
+			if (!(expr is BinaryOperatorExpression {m_Exp1: { }, m_Exp2: { }} binaryExpr)) return false; // binaryExpr has to have both m_Exp1 and m_Exp2 set
+			if ((binaryExpr.m_Operator & BinaryOperatorExpression.RANGES) == 0) return false;
+			
+			binaryExpr.m_Exp2.Compile(bc); // end
+
+			if (binaryExpr.m_Operator == Operator.RightExclusiveRange || binaryExpr.m_Operator == Operator.ExclusiveRange) // ..<, >..< -> dec top of stack
+			{
+				bc.Emit_Literal(DynValue.MinusOne);
+				bc.Emit_Operator(OpCode.Add);
+			}
+			
+			new LiteralExpression(lcontext, DynValue.One).Compile(bc); // step
+			
+			binaryExpr.m_Exp1.Compile(bc); // start
```

Comment:
Can check if m_Exp1 is a number literal (TryGetLiteral) and fold in the plus one

Reply:
Folded in 8e870ee